### PR TITLE
Add support for store access type infomation

### LINKED ISSUE #191: @typescript-eslint/no-unsafe-call with svelte-i18n and @typescript-eslint/recommended-requiring-type-checking
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.20.0

### What version of `eslint-plugin-svelte` are you using?

2.2.0

### What did you do?

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parser: '@typescript-eslint/parser',
  extends: [
    'plugin:svelte/recommended',
    'eslint:recommended',
    'plugin:@typescript-eslint/recommended-requiring-type-checking'
  ],
  parserOptions: {
    project: ['./tsconfig.json'],
    extraFileExtensions: ['.svelte']
  },
  overrides: [
    {
      files: ['*.svelte'],
      parser: 'svelte-eslint-parser',
      parserOptions: {
        parser: "@typescript-eslint/parser",
      },
    },
  ],
  ignorePatterns: ['node_modules', 'dist']
}
```
</details>

```svelte
<script lang="ts">
  import { _ } from 'svelte-i18n';
</script>

<main>
  <input name={$_('test')}>
</main>
```

### What did you expect to happen?

No lint issues to occur.

### What actually happened?

> Unsafe call of an `any` typed value. [@typescript-eslint/no-unsafe-call](https://typescript-eslint.io/rules/no-unsafe-call)

for `$_('test')`. This doesn't happen with https://github.com/sveltejs/eslint-plugin-svelte3.

### Link to Minimal Reproducible Example

https://github.com/binarious/eslint-plugin-svelte-repro

### Additional comments

_No response_

### PR DESCRIPTION
**Pull request by ota-meshi**
close #191

### CODE REVIEW

File: src/context/script-let.ts [hunk 1]
```diff
@@ -528,6 +528,26 @@ export class ScriptLetContext {
...
+    const reservedNames = new Set<string>([
+      "$$props",
+      "$$restProps",
+      "$$slots",
+    ]);
+    for (const nm of maybeStores) {
+      if (reservedNames.has(nm)) continue;
+
+      this.appendScriptWithoutOffset(
+        `declare let $${nm}: Parameters<Parameters<(typeof ${nm})["subscribe"]>[0]>[0];`,
```
Comment (JounQin) [inline src/context/script-let.ts:541]: What if there is already an identifier named `$nm` or it's already declared by the user manually?

This could be uncommon, but before this PR this could happen if the user want to workaround this issue, then this PR would break that.

**Review by JounQin — CHANGES_REQUESTED** [review]

File: src/context/script-let.ts (hunk 1 above)
Comment (ota-meshi) [reply src/context/script-let.ts:541]: I checked the edge case and it didn't seem to be a problem.
https://github.com/ota-meshi/svelte-eslint-parser/blob/store-type/tests/fixtures/parser/ast/ts-store01-type-output.svelte

Also, the svelte compiler doesn't seem to allow you to declare variables that start with $ yourself.



image

```svelte
<script>
	let name = 'world';
	const $n = name
</script>

<h1>Hello {name}!</h1>
```

Comment (JounQin) [reply src/context/script-let.ts:541]: Does `script` with `lang="ts"` + `declare const $n = xxx` works the same?

Comment (ota-meshi) [reply src/context/script-let.ts:541]: I added a test case to check that edge case. That doesn't seem to be a problem either 😉.

https://github.com/ota-meshi/svelte-eslint-parser/blob/store-type/tests/fixtures/parser/ast/ts-store02-type-output.svelte

Comment (JounQin) [reply src/context/script-let.ts:541]: Wow, that is a bit of surprise to me

https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEYD2A7AzgF3gEigLnkxgEsUBzAKFElgWXS1wKNMqA

Comment (JounQin) [reply src/context/script-let.ts:541]: OK, it should due to

```ts
tokens.length = 0
comments.length = 0
removeAllScope(node, result)
```

I just learned a lot from this PR!
